### PR TITLE
Blog-app : Authorization

### PR DESCRIPTION
**Pull request by tajulafreen**
**In this PR I made the following changes:**

- Install CanCanCan in your project.
- Add a role column to the users table. Remember to use a migration for this.
- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- For that you need to implement the post-deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    user ||= User.new # Guest user (not logged in)
+
+    # Define abilities for different roles
+    if user.admin?
+      can :manage, :all # Admins can manage all resources
+    else
+      can :read, :all
+      can :create, Post
+      can :destroy, Post, author_id: user.id
+      can :destroy, Comment, author_id: user.id
```
Comment (VICTOR-CHUKWU) [inline app/models/ability.rb]: - [ ] good job in assigning roles to users, but from this your role, a user can not create a comment, to make it cleaner you can write like this 
```
can :manage, Post, author: user 
         can :manage, Comment, author: user
         can :read, :all
```
here a user can manage comments and posts if he is the owner and also read all

File: app/models/user.rb [hunk 2]
```diff
@@ -16,4 +16,8 @@ class User < ApplicationRecord
   def recent_posts
     posts.order(created_at: :desc).limit(3)
   end
+
+  def admin?
+    role == 'admin'
+  end
```
Comment (VICTOR-CHUKWU) [inline app/models/user.rb:22]: - [ ] good job creating this method here

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @tajulafreen  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- cancancan ability used to define roles


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/ability.rb (hunk 1 above)
Comment (Whoistolu) [reply app/models/ability.rb]: - Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @tajulafreen and @porag-m06 

Great job implementing the recommended changes by the previous reviewer 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**